storage/innobase/row/row0ins.cc

```diff
@@ -1568,6 +1573,74 @@ class ib_dec_in_dtor {
 	ulint&		counter;
 };
 
+/*********************************************************************//**
+Reads sys_trx_end field from clustered index row.
+@return trx_id_t */
+static
+trx_id_t
+row_ins_get_sys_trx_end(
+/*===================================*/
+			const rec_t *rec,	/*!< in: clustered row */
+			ulint *offsets,		/*!< in: offsets */
+			dict_index_t *index)	/*!< in: clustered index */
+{
+	ut_a(dict_index_is_clust(index));
+
+	ulint len;
+	ulint nfield = dict_col_get_clust_pos(
+		&index->table->cols[index->table->vers_end], index);
```

Comment:
This could be a potentially frequent operation. Maybe `vers_end` and `vers_start` should refer directly to clustered index fields instead of referring to table columns.

Reply:
This is local to `row_ins_check_foreign_constraint()`, so it is enough to get `dict_col_t *` on stack before the loop.

Reply:
This is local to `row_ins_check_foreign_constraint()`, so it is enough to get `nfield` on stack before the loop.